sdk/search/azure-search/src/test/java/com/azure/search/implementation/GeoPointDeserializerTests.java

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.implementation;
+
+import com.azure.search.models.GeoPoint;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class GeoPointDeserializerTests {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @BeforeAll
+    public static void setupClass() {
+        SimpleModule module = new SimpleModule();
+        UntypedObjectDeserializer defaultDeserializer = new  UntypedObjectDeserializer(null, null);
+        module.addDeserializer(Object.class, new GeoPointDeserializer(defaultDeserializer));
+        MAPPER.registerModule(module);
+    }
+
+    @Test
+    public void deserializesSimpleGeoJSON() throws Exception {
+        String input = "{"
+            + "\"type\":\"Point\","
+            + "\"coordinates\":[-122.131577,47.678581]"
+            + "}";
+
+        GeoPoint expected = GeoPoint.create(47.678581, -122.131577);
+        Object actual = MAPPER.readValue(input, Object.class);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void deserializesGeoJSONwithCrs() throws Exception {
+        String input = "{"
+            + "\"type\":\"Point\","
+            + "\"coordinates\":[-122.131577,47.678581],"
+            + "\"crs\":{"
+                + "\"type\":\"name\","
+                + "\"properties\":{"
+                    + "\"name\":\"EPSG:4326\""
+                    + "}"
+            + "}}";
+
+        GeoPoint expected = GeoPoint.create(47.678581, -122.131577);
+        Object actual = MAPPER.readValue(input, Object.class);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void returnsDynamicOnInvalidGeoJSON() throws Exception {
+        String input = "{"
+            + "\"type\":\"INVALID_POINT\","
+            + "\"coordinates\":[-122.131577,47.678581]"
+            + "}";
+
+        GeoPoint unexpected = GeoPoint.create(47.678581, -122.131577);
+        Object actual = MAPPER.readValue(input, Object.class);
+
+        Assert.assertNotEquals(unexpected, actual);
+        Assert.assertEquals(LinkedHashMap.class, actual.getClass());
```

Comment:
How does this failure mode manifest out of the top-level APIs like `search`? Does this at some point become an exception, or does it just fail silently?

I guess since during deserialization we don't have the expected type available, the earliest this could throw would be in the process of converting the property bag into a strongly-typed object (at least that's how it works in .NET).

Reply:
yes. This is a custom type that can be used during the convertion to typed objects. This expection will present during that time/